tags/TagContext.md

```diff
@@ -0,0 +1,43 @@
+# Tag Context API
+
+TODO(sebright): Add more detail to the Tag Context summary, including how it
+relates to the current context.
+
+`TagContext` is an abstract data type that represents a map of tags.  It can be
+used to label anything that is associated with a specific operation.  Keys are
+called `TagKey`s, and values are called `TagValue`s.  `TagContext` is serializable,
+and it represents all of the information that must be propagated across process
+boundaries.
+
+## TagKey
```

Comment:
Shall we mention that tag keys are unique within a context, but tag values can be the same?